programs/loader-v4/src/lib.rs

```diff
@@ -412,6 +378,18 @@ pub fn process_instruction_truncate(
     Ok(())
 }
 
+fn find_program_in_cache(
```

Comment:
How about hoisting this method into `InvokeContext`?
All the loader seem to be using it.

Reply:
Yes, that's a good idea.